packages/@o3r/rules-engine/src/engine/operator/operators/basic.operators.ts

```diff
@@ -91,7 +91,27 @@ export const isUndefined: UnaryOperator<any> = {
   evaluator: (input) => input === undefined || input === null
 };
 
+/**
+ * Check if the text variable matches the specified RegExp pattern
+ *
+ * @title matches the pattern
+ */
+export const matchesPattern: Operator<string, string> = {
+  name: 'matchesPattern',
+  evaluator: (value, inputString) => {
+    if (inputString.startsWith('/')) {
+      const finalSlash = inputString.lastIndexOf('/');
+      const regexpPattern = inputString.slice(1, finalSlash);
+      const regexpFlags = inputString.slice(finalSlash + 1);
+      return RegExp(regexpPattern, regexpFlags).test(value);
+    }
```

Comment:
this logic seems to be duplicated 3 times, might be good to extract the code in an helper
```typescript
const inputRegExp = parseRegExp(inputString);
```
or something